config/init_server_creds.go

```diff
@@ -695,14 +695,34 @@ func GeneratePEM(dir string) (key jwk.Key, err error) {
 		}
 		keyFile, err = os.CreateTemp(dir, filenamePattern)
 		if err != nil {
-			err = errors.Wrap(err, "failed to remove temp file")
+			err = errors.Wrap(err, "failed to create private key file")
 			return
 		}
 		fname = keyFile.Name()
 		log.Debugln("Generating new private key in the IssuerKeys directory at", fname)
 	}
 	defer keyFile.Close()
 
+	user, err := GetPelicanUser()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get pelican user for ownership: %v", err)
+	}
+
+	// Windows does not have "chown", has to work differently
+	currentOS := runtime.GOOS
+	if currentOS == "windows" {
+		cmd := exec.Command("icacls", fname, "/grant", user.Username+":F")
+		output, err := cmd.CombinedOutput()
+		if err != nil {
```

Comment:
I think we have a growing number of places in `config` where we re-implement this `if runtime.GOOS == "windows` logic. Can you look around to confirm, and maybe create a helper function for them?